<a href="https://colab.research.google.com/github/DevisriprasadSoma/MLflyrank/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/DevisriprasadSoma/MLflyrank/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### Lane 2 — Refresh / Content Opportunity Scoring

I choose the Refresh / Content Opportunity Scoring lane.

My goal is to identify which content pages should be reviewed first for refresh, improvement, protection, pruning, or monitoring. I chose this lane because the starter dataset contains observable signals such as impressions, clicks, CTR, average position, content age, freshness, and engagement.

These signals can help prioritize pages for human review instead of treating every page equally. I want to investigate whether a simple ranking or ML model can identify useful review candidates better than a fixed hand-written rule.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question

Which content pages should be reviewed first for possible refresh or improvement, based on observable search and engagement signals?

**Decision:** Decide which pages deserve limited review capacity first.

**Action:** A content or SEO reviewer can inspect the highest-ranked pages and decide whether to refresh, expand, protect, prune, or monitor them.

**Cost of a wrong recommendation:** A false positive could waste review and editorial time on a page that does not need attention. A false negative could cause a potentially important or declining page to be missed. Therefore, the goal is to improve the order in which pages are reviewed rather than guarantee that a page will recover.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [15]:
!git clone https://github.com/flyrank-bih/flyrank-ml-internship-starter.git

Cloning into 'flyrank-ml-internship-starter'...
remote: Enumerating objects: 283, done.
remote: Counting objects: 100% (145/145), done.
remote: Compressing objects: 100% (62/62), done.
remote: Total 283 (delta 112), reused 83 (delta 83), pack-reused 138 (from 1)
Receiving objects: 100% (283/283), 1.85 MiB | 4.73 MiB/s, done.
Resolving deltas: 100% (153/153), done.


In [16]:
import os

os.chdir("/content/flyrank-ml-internship-starter")

print(os.getcwd())
print(os.path.exists("data/raw/content_refresh_anonymized.csv"))

/content/flyrank-ml-internship-starter
True


In [17]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))
print("Number of columns:", len(df.columns))

declining_rate = (
    df["trend_direction"].str.lower() == "down"
).mean()

print("Declining-page rate:", round(declining_rate, 3))

visible_pages = (df["impressions_90d"] >= 500).sum()

print("Pages with 500+ impressions:", visible_pages)

Number of pages: 30000
Number of columns: 44
Declining-page rate: 0.542
Pages with 500+ impressions: 16726


### Observation

The starter dataset contains 30,000 pages across 44 columns. About 54.2% of the pages are currently classified as declining based on the starter `trend_direction` label. Also, 16,726 pages have at least 500 impressions over the 90-day period.

These numbers suggest that there is a meaningful review-prioritization problem: many pages have enough search exposure to potentially matter, while a substantial share show a declining trend. This makes it worth investigating whether observable page-level signals can help prioritize which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can claim

I can identify observed relationships between page-level search, engagement, freshness, and content signals and use them to prioritize pages for human review.

If a ranking or ML model performs better than a simple baseline under a suitable validation setup, I can report that result for the data and evaluation used.

### What I cannot claim

I cannot claim that a particular feature causes a change in search performance or that I have discovered a Google ranking factor.

I also cannot claim that refreshing a page will definitely make it recover.

The starter `trend_direction` label is based on the current data window, so I will treat it as a starting point for exploration rather than automatically treating it as a future outcome.

The goal is decision support and prioritization, not an automatic guarantee of business results.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled
- [x] Notebook runs top to bottom
- [x] No client names, URLs, private queries
- [x] Claims use careful words
- [x] Committed to repo